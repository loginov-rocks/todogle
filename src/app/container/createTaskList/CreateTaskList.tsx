import { Button, TextField } from '@material-ui/core';
import * as React from 'react';

import useInput from '../../../hooks/useInput';

import gapi from '../../../services/gapi';
import { TaskListResource } from '../../../services/gapi/TaskListResource';

interface Props {
  onCreate: (taskList: TaskListResource) => void;
}

const CreateTaskList = ({ onCreate }: Props) => {
  const { value, bindProps, reset } = useInput('');

  const handleSubmit = (event: any) => {
    event.preventDefault();

    gapi.createTaskList({ title: value })
      .then(taskList => {
        onCreate(taskList);
        reset();
      });
  };

  return (
    <form onSubmit={handleSubmit}>
      <TextField required {...bindProps} />
      <Button type="submit" variant="contained">Create Task List</Button>
    </form>
  );
};

export default CreateTaskList;
