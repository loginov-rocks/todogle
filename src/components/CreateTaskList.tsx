import * as React from 'react';

import gapi from '../services/gapi';
import { TaskListResource } from '../services/gapi/TaskListResource';

interface Props {
  onCreate: (taskList: TaskListResource) => void;
}

interface State {
  title: string;
}

export default class CreateTaskList extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      title: '',
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event: any) {
    this.setState({ title: event.target.value });
  }

  handleSubmit(event: any) {
    event.preventDefault();

    const { onCreate } = this.props;
    const { title } = this.state;

    gapi.createTaskList({ title })
      .then(taskList => {
        onCreate(taskList);
        this.setState({ title: '' });
      });
  }

  render() {
    const { title } = this.state;

    return (
      <form onSubmit={this.handleSubmit}>
        <input onChange={this.handleChange} value={title} />
        <button type="submit">Create Task List</button>
      </form>
    );
  }
}
