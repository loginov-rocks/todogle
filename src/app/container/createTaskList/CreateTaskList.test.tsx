import { shallow } from 'enzyme';
import * as React from 'react';

import CreateTaskList from './CreateTaskList';

it('matches snapshot', () => {
  const wrapper = shallow(
    <CreateTaskList onCreate={() => undefined} />,
  );

  expect(wrapper).toMatchSnapshot();
});
