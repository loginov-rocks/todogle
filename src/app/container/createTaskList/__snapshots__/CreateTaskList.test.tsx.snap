// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <WithStyles(ForwardRef(TextField))
    onChange={[Function]}
    required={true}
    value=""
  />
  <WithStyles(ForwardRef(Button))
    type="submit"
    variant="contained"
  >
    Create Task List
  </WithStyles(ForwardRef(Button))>
</form>
`;
